service/history/mutableStateBuilder.go

```diff
@@ -2302,6 +2297,8 @@ func (e *mutableStateBuilder) AddContinueAsNewEvent(decisionCompletedEventID int
 	}
 
 	continueAsNewEvent := e.hBuilder.AddContinuedAsNewEvent(decisionCompletedEventID, newRunID, attributes)
+	// Write the event to cache only on active cluster
+	e.writeCompletionEventToCache(continueAsNewEvent)
```

Comment:
I think we should move writing event to cache logic to Replicate*** API within mutable state for all closed events.
Then you can remove this comment too.

Reply:
Updated